src/storage/mvcc/reader/scanner/entry.rs

```diff
@@ -0,0 +1,700 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! A dedicate scanner that outputs content in each CF.
+
+use std::cmp::Ordering;
+
+use engine::CF_DEFAULT;
+use kvproto::kvrpcpb::IsolationLevel;
+
+use crate::storage::kv::SEEK_BOUND;
+use crate::storage::mvcc::write::{Write, WriteType};
+use crate::storage::mvcc::Result;
+use crate::storage::txn::TxnEntry;
+use crate::storage::{Cursor, Key, Lock, Snapshot, Statistics};
+
+use super::util::CheckLockResult;
+use super::ScannerConfig;
+
+/// This struct can be used to scan keys starting from the given user key (greater than or equal).
+///
+/// Internally, for each key, rollbacks are ignored and smaller version will be tried. If the
+/// isolation level is SI, locks will be checked first.
+///
+/// Use `ScannerBuilder` to build `EntryScanner`.
+///
+/// Note: The implementation is almost the same as `ForwardScanner`, made a few
```

Comment:
What's the difference specifically? Is the difference only limited to, in forward scanner we throw out KVs and here we throw out TxnEntries?

Reply:
Yes, but later it needs to support incremental scan.

Reply:
Could you further explain what is incremental scan? I'm not familiar with the background. Current Mvcc scanners work like an iterator, can be paused and continued at any time before nothing left.

Reply:
The difference is how to scan data, point in time vs time range(incremental scan).

Reply:
Got it.